optimum/habana/transformers/models/baichuan/quantizer.py

```diff
@@ -0,0 +1,237 @@
+# Copyright 2023 Baichuan Inc. All Rights Reserved.
+
+# Copyright 2022 EleutherAI and the HuggingFace Inc. team. All rights reserved.
+#
+# This code is based on EleutherAI's GPT-NeoX library and the GPT-NeoX
+# and OPT implementations in this library. It has been modified from its
+# original forms to accommodate minor architectural differences compared
+# to GPT-NeoX and OPT used by the Meta AI team that trained the model.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Adapted from the following sources:
+https://huggingface.co/baichuan-inc/Baichuan2-7B-Chat/blob/main/quantizer.py
+https://huggingface.co/baichuan-inc/Baichuan2-13B-Chat/blob/main/quantizer.py
+"""
+
+import bitsandbytes as bnb
+from bitsandbytes.nn.modules import Params4bit, Int8Params
+import torch
+
+def Params4bitCuda(self, device):
```

Comment:
cuda? are 4bit and 8bit of baichuan supported, I see you import bitsandbytes. is bitsandbytes support in habana?

Reply:
Thanks for pointing it out. It doesn't support INT8/4 quantization for baichuan in habana.  I removed the stale code in the new version.